backend/src/receipts.js

```diff
@@ -57,10 +93,99 @@ async function getReceiptsList(blockHash) {
       kind: indexerCompatibilityTransactionActionKinds.get(receiptAction.kind),
     });
   }
+  return receipts;
+}
 
+async function getExucutedReceiptsList(blockHash) {
+  const receiptActions = await queryExecutedReceiptsList(blockHash);
+  // The receipt actions are ordered in such a way that the actions for a single receipt go
+  // one after another in a correct order, so we can collect them linearly using a moving
+  // window based on the `previousReceiptId`.
+  let receipts = [];
+  let actions;
+  let previousReceiptId = "";
+  const indexerCompatibilityTransactionActionKinds = await getIndexerCompatibilityTransactionActionKinds();
+  for (const receiptAction of receiptActions) {
+    if (previousReceiptId !== receiptAction.receipt_id) {
+      previousReceiptId = receiptAction.receipt_id;
+      actions = [];
+      const receipt = {
+        actions,
+        blockTimestamp: new BN(receiptAction.executed_in_block_timestamp)
+          .divn(10 ** 6)
+          .toNumber(),
+        gasBurnt: receiptAction.gas_burnt,
+        receiptId: receiptAction.receipt_id,
+        receiverId: receiptAction.receiver_id,
+        signerId: receiptAction.predecessor_id,
+        status: INDEXER_COMPATIBILITY_RECEIPT_ACTION_KINDS.get(
+          receiptAction.status
+        ),
+        originatedFromTransactionHash:
+          receiptAction.originated_from_transaction_hash,
+        tokensBurnt: receiptAction.tokens_burnt,
+      };
+      receipts.push(receipt);
+    }
+    actions.push({
+      args: receiptAction.args,
+      kind: indexerCompatibilityTransactionActionKinds.get(receiptAction.kind),
+    });
+  }
```

Comment:
This part is just for demo and if we'll leave this implementation I'll refactor it